### PR TITLE
journal and article route and middleware

### PR DESCRIPTION
**Pull request by ghimirear**
con you try it again?

### CODE REVIEW

File: client/src/components/Header/Header.js [hunk 1]
```diff
@@ -1,9 +1,27 @@
...
+const Header = ()=> {
+    const[user, setUser] = useState(JSON.parse(localStorage.getItem("user")));
+
+    const logout = ()=>{
+        localStorage.clear();
+        window.location.href="/login";
+    }
     return(
         <nav class="navbar navbar-dark bg-primary">
-            <div className="row col-12 d-flex justify-content-center text-white">
```
Comment (ghimirear) [inline client/src/components/Header/Header.js:5]: ag

File: client/src/components/Header/Header.js [hunk 2]
```diff
@@ -1,9 +1,27 @@
...
+// import {Redirect, useHistory} from "react-router-dom";
+const Header = ()=> {
+    const[user, setUser] = useState(JSON.parse(localStorage.getItem("user")));
+
+    const logout = ()=>{
+        localStorage.clear();
+        window.location.href="/login";
+    }
     return(
         <nav class="navbar navbar-dark bg-primary">
```
Comment (ghimirear) [inline client/src/components/Header/Header.js:13]: ag

File: client/src/components/Home/Home.js [hunk 3]
```diff
@@ -1,8 +1,57 @@
...
+            console.log(res)
+        )
+        .catch(err =>{
+            console.log(err)
+        })
+    }
+
     return(
         <div className="mt-2">
-            Home page content
```
Comment (ghimirear) [inline client/src/components/Home/Home.js:5]: ag

File: middleware/auth.js [hunk 4]
```diff
@@ -1,15 +1,22 @@
-import jwt from "jsonwebtoken";
```
Comment (ghimirear) [inline middleware/auth.js:1]: ag

File: middleware/auth.js [hunk 5]
```diff
@@ -1,15 +1,22 @@
...
+const {jwt, decode} = require ("jsonwebtoken");
 const auth = async (req, res, next)=>{
+  console.log(req.headers);
 try {
     const token = req.headers.authorization.split(" ")[1];
+    const isCustomAuth = token.length < 500;
+
     let decodedData;
-    if(token){
```
Comment (ghimirear) [inline middleware/auth.js:6]: ag

Comment (ghimirear) [inline middleware/auth.js:6]: ag

File: middleware/auth.js [hunk 6]
```diff
@@ -1,15 +1,22 @@
...
+    else{
+      decodedData= jwt.decode(token);
+      req.userId = decodedData?.sub;
+    }
     next();
 } catch (error) {
   console.log(error)  
 }
 }
-export default auth;
```
Comment (ghimirear) [inline middleware/auth.js:15]: ag

File: models/Article.js [hunk 7]
```diff
@@ -10,8 +10,8 @@ const articleSchema = new Schema({
       required : true
     },
    // each article belong to one user and one Journal so it is not an array
-    user:{ type:Schema.Types.ObjectId, ref :"User"},
```
Comment (ghimirear) [inline models/Article.js:13]: ag

Comment (ghimirear) [inline models/Article.js:13]: ag

Comment (ghimirear) [inline models/Article.js:13]: ag

File: models/Journal.js [hunk 8]
```diff
@@ -8,7 +8,7 @@ const journalSchema = new Schema({
       required : true
     },
    
-    user:{ type:Schema.Types.ObjectId, ref :"User"},
```
Comment (ghimirear) [inline models/Journal.js:11]: ag

File: models/Quote.js [hunk 9]
```diff
@@ -6,7 +6,7 @@ const quoteSchema = new Schema({
       type: String,
     },
     // each quote belongs to single user
-    user:{ type:Schema.Types.ObjectId, ref :"User"}
```
Comment (ghimirear) [inline models/Quote.js:9]: ag

File: server.js [hunk 10]
```diff
@@ -3,21 +3,30 @@ var cors = require('cors');
...
 
  app.use(cors());
+ app.use(cookieParser());
 
 // Serve up static assets (usually on heroku)
 if (process.env.NODE_ENV === "production") {
   app.use(express.static("client/build"));
 }
 app.use("/api", userRoutes);
-
```
Comment (ghimirear) [inline server.js:20]: ag

**Review by jenneration — APPROVED** [review]

**Review by ghimirear — COMMENTED** [review]
reviewed

**Review by jenneration — APPROVED** [review]

**Review by jenneration — APPROVED** [review]
ok

**Review by jenneration — APPROVED** [review]
ok

File: client/src/components/Header/Header.js [hunk 11]
```diff
@@ -1,9 +1,27 @@
-import React from 'react';
```
Comment (jenneration) [inline client/src/components/Header/Header.js:1]: ok

Comment (jenneration) [inline client/src/components/Header/Header.js:1]: ok

File: client/src/components/Header/Header.js [hunk 12]
```diff
@@ -1,9 +1,27 @@
...
-function Header() {
```
Comment (jenneration) [inline client/src/components/Header/Header.js:2]: ok

File: client/src/components/Header/Header.js (hunk 1 above)
Comment (jenneration) [inline client/src/components/Header/Header.js:5]: ok

File: client/src/components/Header/Header.js [hunk 13]
```diff
@@ -1,9 +1,27 @@
...
+    const[user, setUser] = useState(JSON.parse(localStorage.getItem("user")));
+
+    const logout = ()=>{
+        localStorage.clear();
+        window.location.href="/login";
+    }
     return(
         <nav class="navbar navbar-dark bg-primary">
-            <div className="row col-12 d-flex justify-content-center text-white">
-            <span className="h3">Register</span>
```
Comment (jenneration) [inline client/src/components/Header/Header.js:6]: ok

File: client/src/components/Home/Home.js [hunk 14]
```diff
@@ -1,8 +1,57 @@
-import React from 'react';
```
Comment (jenneration) [inline client/src/components/Home/Home.js:1]: ok

File: client/src/components/Home/Home.js [hunk 15]
```diff
@@ -1,8 +1,57 @@
...
-function Home(props) {
```
Comment (jenneration) [inline client/src/components/Home/Home.js:2]: ok

File: client/src/components/Home/Home.js (hunk 3 above)
Comment (jenneration) [inline client/src/components/Home/Home.js:5]: ok

File: middleware/auth.js (hunk 4 above)
Comment (jenneration) [inline middleware/auth.js:1]: ok

File: middleware/auth.js (hunk 5 above)
Comment (jenneration) [inline middleware/auth.js:6]: ok

File: middleware/auth.js (hunk 6 above)
Comment (jenneration) [inline middleware/auth.js:15]: ok

File: models/Article.js (hunk 7 above)
Comment (jenneration) [inline models/Article.js:13]: ok

File: models/Article.js [hunk 16]
```diff
@@ -10,8 +10,8 @@ const articleSchema = new Schema({
...
-    journal:{type:Schema.Types.ObjectId, ref :"Journal"}
```
Comment (jenneration) [inline models/Article.js:14]: ok

File: models/Journal.js (hunk 8 above)
Comment (jenneration) [inline models/Journal.js:11]: ok

File: models/Quote.js (hunk 9 above)
Comment (jenneration) [inline models/Quote.js:9]: ok

File: server.js (hunk 10 above)
Comment (jenneration) [inline server.js:20]: ok

**Review by jenneration — APPROVED** [review]
ok

File: client/src/components/Header/Header.js (hunk 11 above)
Comment (ghimirear) [inline client/src/components/Header/Header.js:1]: ag

File: client/src/components/Header/Header.js (hunk 1 above)
Comment (ghimirear) [inline client/src/components/Header/Header.js:5]: ag

File: client/src/components/Header/Header.js (hunk 13 above)
Comment (ghimirear) [inline client/src/components/Header/Header.js:6]: ag

File: client/src/components/Header/Header.js (hunk 11 above)
Comment (jenneration) [inline client/src/components/Header/Header.js:1]: ok

File: client/src/components/Header/Header.js (hunk 12 above)
Comment (jenneration) [inline client/src/components/Header/Header.js:2]: ok

File: client/src/components/Header/Header.js (hunk 1 above)
Comment (jenneration) [inline client/src/components/Header/Header.js:5]: ok

File: client/src/components/Header/Header.js (hunk 13 above)
Comment (jenneration) [inline client/src/components/Header/Header.js:6]: ok

File: client/src/components/Home/Home.js (hunk 14 above)
Comment (jenneration) [inline client/src/components/Home/Home.js:1]: ok

File: client/src/components/Home/Home.js (hunk 15 above)
Comment (jenneration) [inline client/src/components/Home/Home.js:2]: ok

File: client/src/components/Home/Home.js (hunk 3 above)
Comment (jenneration) [inline client/src/components/Home/Home.js:5]: ok

File: middleware/auth.js (hunk 4 above)
Comment (jenneration) [inline middleware/auth.js:1]: ok

File: middleware/auth.js (hunk 5 above)
Comment (jenneration) [inline middleware/auth.js:6]: ok

File: middleware/auth.js (hunk 6 above)
Comment (jenneration) [inline middleware/auth.js:15]: ok

File: models/Article.js (hunk 7 above)
Comment (jenneration) [inline models/Article.js:13]: ok

File: models/Article.js (hunk 16 above)
Comment (jenneration) [inline models/Article.js:14]: ok

File: models/Journal.js (hunk 8 above)
Comment (jenneration) [inline models/Journal.js:11]: ok

File: models/Quote.js (hunk 9 above)
Comment (jenneration) [inline models/Quote.js:9]: ok

File: server.js (hunk 10 above)
Comment (jenneration) [inline server.js:20]: ok

**Review by jenneration — APPROVED** [review]
ok

**Comment by jenneration** [comment]
no okee doke

**Comment by jenneration** [comment]
okee doke again

**Pull request merged**